src/controller/python/chip/clusters/idl.py

```diff
@@ -0,0 +1,253 @@
+#
+#    Copyright (c) 2022 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from multiprocessing.sharedctypes import Value
+import os
+
+from lark import Lark
+from chip.idl import matter_idl_parser, matter_idl_types
+from . import ClusterObjects
+from . import Attribute
+import chip.tlv as tlv
+
+import enum
+import dataclasses
+import typing
+
+
+def CreateParser():
+    """
+    Generates a parser that will process a ".matter" file into a IDL
+    """
+    return Lark.open('../idl/matter_grammar.lark', rel_to=__file__, start='idl', parser='lalr', transformer=matter_idl_parser.MatterIdlTransformer())
```

Comment:
this path does not seem to exist: ../idl is not a folder but rather something in scripts.